### PR TITLE
Added debug mode for canary test pipeline

### PR DESCRIPTION
**Pull request by sanjuyadav24**
### **Context**
[AB#2328157](https://mseng.visualstudio.com/AzureDevOps/_workitems/edit/2328157)

---

### **Task Name**
NA

---

### **Description**
Update canarytest pipeline to use node version from task and run in debug mode

---

### **Risk Assessment** (Low / Medium / High)  
Low

---

### **Change Behind Feature Flag** (Yes / No)
No

---

### **Tech Design / Approach**
NA

---

### **Documentation Changes Required** (Yes/No)
NA

---

### **Unit Tests Added or Updated** (Yes / No)  
NA

---

### **Additional Testing Performed**
NA

---

### **Logging Added/Updated** (Yes/No)
NA

--- 

### **Telemetry Added/Updated** (Yes/No) 
NA

---

### **Rollback Scenario and Process** (Yes/No)
NA

---

### **Dependency Impact Assessed and Regression Tested** (Yes/No)
NA

---

### CODE REVIEW

File: ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts [hunk 1]
```diff
@@ -29,3 +29,65 @@ export function getBuildConfigs(task: string): string[] {
...
+        const taskJson = JSON.parse(fs.readFileSync(taskJsonPath, 'utf8'));
+        const execution = taskJson.execution || {};
+        
+        // Extract all Node versions from execution handlers
+        const nodeVersions: number[] = [];
+        
+        for (const handler in execution) {
+            if (handler.startsWith('Node')) {
+                // Extract number from handler name (e.g., Node10 -> 10, Node20_1 -> 20, Node24 -> 24)
+                const match = handler.match(/Node(\d+)/);
```
Comment (rishabhmalikMS) [inline ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts:57]: What would happen in node 6 cases here for tasks which have old node 6 handler. Ask the handler has "node" and not "node6" if am not wrong. We would not retrieve any number in that case from this regex. I understand this is a corner case for older tasks.

File: ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts [hunk 2]
```diff
@@ -29,3 +29,65 @@ export function getBuildConfigs(task: string): string[] {
...
+        }
+        
+        // Return highest Node version
+        if (nodeVersions.length > 0) {
+            const maxVersion = Math.max(...nodeVersions);
+            console.log(`Found Node versions for ${buildConfig || taskName}: [${nodeVersions.join(', ')}], using highest: ${maxVersion}`);
+            return maxVersion;
+        }
+        
+        console.log(`No Node execution handlers found in ${taskJsonPath}`);
```
Comment (rishabhmalikMS) [inline ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts:71]: Continuing from above comment, if only node 6 handler is there, it would return null instead of node 6.

File: ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts (hunk 1 above)
Comment (sanjuyadav24) [reply ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts:57]: we dont have any task with node6 handler, its already deprecated
we have node6 support only in agent for custom tasks and extensions

File: ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts (hunk 2 above)
Comment (sanjuyadav24) [reply ci/ci-test-tasks/test-and-verify-v2/src/helpers.ts:71]: we dont have any task with node6 handler, its already deprecated
we have node6 support only in agent for custom tasks and extensions

File: ci/ci-test-tasks/test-and-verify-v2/src/index.ts [hunk 3]
```diff
@@ -198,21 +198,43 @@ async function runTaskPipelines(taskName: string, pipeline: BuildDefinitionRefer
...
+  } else if (envNodeVersion) {
+    console.log(`Using Node version ${nodeVersion} from environment variable for task ${taskName}`);
+  } else {
+    throw new Error(`Cannot determine Node version for task ${taskName}. Neither task.json nor CANARY_TEST_NODE_VERSION environment variable provided.`);
+  }
+
+  // Enable debug mode for triggered pipelines
+  const debugMode = process.env.CANARY_TEST_DEBUG_MODE;
+  const buildParameters: Record<string, string> = {
+    CANARY_TEST_TASKNAME: pipeline.name!,
```
Comment (rishabhmalikMS) [inline ci/ci-test-tasks/test-and-verify-v2/src/index.ts]: We have forceful value assignment here, could you please ensure this pipeline.name does not have any impact if its null, in original code it was not using ! operator.
`CANARY_TEST_TASKNAME: pipeline.name!`

Comment (sanjuyadav24) [reply ci/ci-test-tasks/test-and-verify-v2/src/index.ts]: removed this check
